ui/apps/platform/src/Containers/AccessControl/AuthProviders/AuthProviderForm.tsx

```diff
@@ -57,7 +63,14 @@ export type AuthProviderFormProps = {
     onClickEdit: () => void;
 };
 
-const authProviderState = createStructuredSelector({
+type AuthProviderState = {
+    roles: Role[];
+    groups: Group[];
+    saveAuthProviderStatus: { status: string; message: string } | null;
+    availableProviderTypes: AuthProviderInfo[];
+};
+
+const authProviderState = createStructuredSelector<AuthProviderState>({
```

Comment:
Possible follow up during 4.8A sprint to evaluate pro and con;
* `createStructuredSelector` that requires separate type
* separate `useSelector` hooks which might infer type if reducer is TypeScript

Reply:
It might be a moot point for this form, because ordinary requests seem even better.